Moved contents of test/scripts in radare2 repository into src/ in rad…

valdaarhun opened the pull request:
…are2-testbins repository

<!--- Filling this template is mandatory -->

**Detailed description**

<!-- Explain the **details** for making this change. Is a new feature implemented? What existing problem does the pull request solve? How does the pull request solve these issues? Please provide enough information so that others can review your pull request. -->

...

**Test plan**

<!-- What steps should the reviewer take to test your pull request? Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots/videos. This is your time to re-check that everything works and that you covered all the edge cases -->

...

**Closing issues**

<!-- put "closes #XXXX" in your comment to auto-close the issue that your PR fixes (if any). -->

Might help in resolving issue [#16576](https://github.com/radareorg/radare2/issues/16576) in the `radare2` repository

valdaarhun commented:
@trufae I have moved the contents of test/scripts from the `radare2` repo to the src directory in the `radare2-testbins` repo. Could you please check if this is right?

trufae commented:
Please update the broken tests:

```
$ git grep scripts
README.md: * fuzz/:        Fuzzing helper scripts
db/archos/darwin-x64/http:CMDS=!scripts/test-webserver.sh
db/archos/linux-x64/http:CMDS=!scripts/test-webserver.sh
db/archos/windows-x64/http:CMDS=!scripts\test-webserver.bat
db/cmd/cmd_arp:arpg scripts/gdb-reg-profile.txt > $a
db/cmd/cmd_arp:arpg scripts/gdb-reg-profile-invalid.txt > $b
db/cmd/cmd_interpret:. scripts/r2test.spp
db/cmd/cmd_pd:. scripts/palette.r2
db/cmd/cmd_pipe:#!pipe python3 scripts/get-funcs.py
db/cmd/cmd_print:. scripts/palette.r2
db/cmd/cmd_print:. scripts/palette.r2
db/cmd/cmd_px:. scripts/palette.r2
db/formats/smd:. scripts/smd_strings.r2
db/tools/r2:CMDS=!radare2 -i scripts/no-nl-at-eof.r2 -NQ -
db/tools/rafind2:CMDS=!rafind2 -F scripts/keyword bins/mach0/FileDP
$
```

valdaarhun commented:
I didn't really understand what "update the broken tests" means. I moved the location of the test files so I am guessing I will have to tell the above files where to find the test files, right?
For eg, in db/archos/darwin-x64/http, there is the following line:
`CMDS=!scripts/test-webserver.sh`
So, will changing this location fix the broken test for this particular file?

trufae commented:
yes

trufae commented:
btw if you use the same branch name in here, than your pr in r2.. in theory the testsuite will pick both PRs at once and test it properly. otherwise nothing is tested when commiting here.

@ret2libc can u confirm this behaviour is still the same after r2r was archived?

ret2libc commented:
I don't think this is true anymore. Bins are supposed to be just added in this repository, so in general you can first commit stuff here and then do stuff in r2.

valdaarhun commented:
A new file called `test-webserver.bat` was added to the `/scripts` folder in the `radare2` repository. Should I move that to the `testbins` repository as well?

valdaarhun commented:
@trufae I think I have updated all the broken tests and have made commits in the `radare2` repo. Please let me know if something has been left out.

kazarmy commented:
> I don't think this is true anymore. Bins are supposed to be just added in this repository, so in general you can first commit stuff here and then do stuff in r2.

A problem is that you can't first commit stuff here that would break current r2r, even if you immediately submitted a matching PR in r2, since PRs there are neither immediately approved nor merged. Ofc there are ways to work around this, involving extra (hopefully simple) PRs/commits here and there.

Btw isn't the `src/` folder supposed to be for the source code of test binaries?

valdaarhun commented:
Hi. I am sorry I couldn't get back for the last 3 weeks because of college exams. However, I am free now once again and would like to continue working on this.

valdaarhun commented:
@kazarmy Yeah, if I am not mistaken the `src/` folder contains the source code for the test binaries.

trufae commented:
I'm fine with this change

trufae commented on src/smd_strings.r2:
```diff
@@ -0,0 +1,412 @@
+"wz SEGA GENESIS    (C)SEGA 1995.MARCART. BRADESCO                                  CART. BRADESCO                                  GM 00000001-00W`J               " @ 256
```
i think moving files like this to bins will make it harder to debug and fix bugs in the flag filtering and command parsing. it is better to keep the scripts in r2 repo, and just use the bins/src as a place to keep the source of the programs in there.

valdaarhun replied:
I was going through the files to see which ones I should move back to the r2 repository. For a few files, I am not sure about whether I should move them back to the r2 repo or leave them here. I guess I should move `src/smd_strings.r2`, `src/gdb-reg-profile-invalid.txt`, `src/gdb-reg-profile.txt`, `src/palette.r2` back to the r2 repo. Is that right?

valdaarhun replied:
Would it be right to leave `src/r2test.spp `, `src/test-webserver.sh` and `src/test-webserver.bat` over here in bins/src? Also, I haven't exactly understood what `src/no-nl-at-eof.r2` and `src/keyword` do.

trufae replied:
Yes, those files that depend on r2 commands must go into the r2 repo, the bins repo should only contain binaries. Another issue i have noticed is that projects should be moved too.

valdaarhun replied:
I think I have understood which scripts to move back to the r2 repo. However, I haven't understood which projects you are referring to that need to be moved.

valdaarhun replied:
What do I do if I am not sure whether a particular file depends on r2 commands? Should I leave them here for the time being?

trufae commented:
I'll merge after you update the PR moving those remaining files up and down. that will help me on fixing projects without making master everytime the project format changes and such. Thanks!!

trufae commented:
ping

valdaarhun commented:
Hi. Sorry for the late reply. So far I have been going through the files to try and understand the code. I also went through the ebook and the documentation generated on running the doxygen command. I have been meaning to contact you but I thought I would go through all of this first to accumulate my doubts.

trufae commented:
Just do the scripts, for projects, as long as they are tied to commands, this can change frequently, and the resulting script can be really big for some binaries, so i would keep the projects in r2/test instead of r2/test/bins because they are subject to change and having them in a separate repo makes it more complicated to get master green when commiting fixes for projects.

also, i am even thinking in making all the project tests to not depend on already saved projects, just save them from scratch because i removed the transitional support to convert projects from old "formats" for the sake of simplicity.

So sumarizing, dont touch projects for now, but i think i'll end up removnig them from bins and i dont want them to bload the main repo

valdaarhun commented:
Are there any docs that explain the stuff in files like `smd_strings.r2`? Because I couldn't find any related doc in the ebook or doxygen generated documentation. Also, the link in the modules section of `radare2/DEVELOPERS.md` is broken.

valdaarhun commented:
Ok. So I guess I will just move the scripts for now.

valdaarhun commented:
Hi. I have moved back some of the files from `bins/src/` to `scripts/`. Please let me know if I have moved a file that shouldn't be moved or if I have forgotten to move a file that should be moved.

trufae commented:
im merging, lgtm, lets just hope the ci gets green after this merge  :) thanks for the patience

valdaarhun commented:
Thanks. I enjoyed working on this issue!
